source/daplink/drag-n-drop/vfs_user.c

```diff
@@ -529,20 +542,49 @@ static uint32_t expand_info(uint8_t *buf, uint32_t bufsize)
                 memcpy(buf, insert_string, str_len);
             }else{
                 //stop the string expansion and leave as it is
-                buf += buf_len;
+                buf += buf_len; 
                 break;
             }
-
+						
         }
     } while (*buf != '\0');
-
+		
     return (buf - orig_buf);
 }
 
 // Initialize flash algo, erase flash, uninit algo
 static void erase_target(void)
 {
+#if defined(TARGET_MCU_PSOC6) && defined(DAPLINK_IF)
+	flash_region_t region = FLASH_REGION_SFLASH_TOC2;
+  if (kit_has_sflash_restriction())
+	{
+		region = FLASH_REGION_WFLASH;
+	} 
+
+	while ( 1 ) 
+	{
+		if ( switch_target_flash_algo(region) != ERROR_SUCCESS )
+		{
+			break;
+		}
+		if ( flash_intf_target->erase_chip() != ERROR_SUCCESS && region != FLASH_REGION_SMIF )
+		{	
+			break;
+		}
+		if ( region != FLASH_REGION_MAIN )
+		{
+			region--;
+		}
+		else
+		{
+			break;
+		}
+	} 
+	
+#else	
```

Comment:
Need to insert this logic to flash algo api.

Reply:
this logic has been eliminated